influxdb3_py_api/src/system_py.rs

```diff
@@ -511,6 +514,60 @@ pub fn execute_python_with_batch(
     })
 }
 
+pub fn execute_cron_trigger(
+    code: &str,
+    schedule_time: DateTime<Utc>,
+    schema: Arc<DatabaseSchema>,
+    query_executor: Arc<dyn QueryExecutor>,
+    args: &Option<HashMap<String, String>>,
+) -> PyResult<PluginReturnState> {
+    Python::with_gil(|py| {
+        // import the LineBuilder for use in the python code
+        let globals = PyDict::new(py);
+
+        let py_datetime = PyDateTime::from_timestamp(py, schedule_time.timestamp() as f64, None)?;
+
+        py.run(
+            &CString::new(LINE_BUILDER_CODE).unwrap(),
+            Some(&globals),
+            None,
+        )?;
+
+        let api = PyPluginCallApi {
+            db_schema: schema,
+            query_executor,
+            return_state: Default::default(),
+        };
+        let return_state = Arc::clone(&api.return_state);
+        let local_api = api.into_pyobject(py)?;
+
+        // turn args into an optional dict to pass into python
+        let args = args.as_ref().map(|args| {
```

Comment:
since we also have this block in the `execute_wal_plugin`, can we pull into a function?

Reply:
factored it out.